backend/Application/Aggregates/Contract/Observability/ContractMetrics.cs

```diff
@@ -0,0 +1,94 @@
+using System.Diagnostics;
+using Application.Aggregates.Contract.Types;
+using Prometheus;
+
+namespace Application.Aggregates.Contract.Observability;
+
+internal static class ContractMetrics
```

Comment:
If possible, make this class generic and place it in a class library, so it's not tied to Contracts and so it can be used elsewhere.

Reply:
I think having the metric code for smart contract “close” to other contract code adheres to done elsewhere for contract implementation.

The Duration metric could be made generic but then I would need to make the property ReadDuration non private. I will keep it for now and refactor when more aggregates are added.